### ISSUE TITLE
Add support for new "Spot" instances

### ISSUE THREAD

**Issue by bo-qeye** [doitintl/gcpinstances.info#12]
GCP recently announced an evolution of preemptible VMs. They are now called Spot VMs.

More info here: https://cloud.google.com/compute/docs/instances/spot

**Comment by nchicong** [comment]
+1
Please add a column for Spot/ preemptible.

**Comment by TheKevJames** [comment]
Note that Spot pricing is not (or is no longer) the same as) preemptible pricing.

From https://cloud.google.com/compute/docs/instances/spot#pricing
> Spot prices, the prices for Spot VMs, change over time, up to once every 30 days. Specifically, Spot VMs provide at least a 60% discount and up to a 91% discount on the price of standard VMs with the same machine types.

We would need to scrape currently Spot pricing ~monthly and add an entirely new column, as it is valuable to see standard, preemptible, _and_ spot rates.

Note that the [catalog API](https://cloud.google.com/billing/v1/how-tos/catalog-api) does now support spot pricing in their results.

**Comment by Burekasim** [comment]
Hi Kevin, 

Thanks for letting me know, when I started working on this website, it took me 4 days until I had to abandon the catalog api because there was no way to get to the same price of gcp calculator. 
I will have to review this again. maybe I will start with the spot prices and we will figure out the rest in a later date.

**Comment by TheKevJames** [comment]
Yeah, I haven't played with the catalog API yet, but it looked... more complicated than I'd prefer...

I notice (some of?) the GCP pricing page docs have been updated to include the Spot pricing as well (and in some cases no longer show preemptible), so there might be some scraping that can be done?

Mostly the important thing IMO is that the Spot pricing changes monthly and varies per-SKU, so #34 might end up being a prerequisite here :(

**Comment by gdubicki** [comment]
> Thanks for letting me know, when I started working on this website, it took me 4 days until I had to abandon the catalog api because there was no way to get to the same price of gcp calculator.

> Yeah, I haven't played with the catalog API yet, but it looked... more complicated than I'd prefer...

I also considered using the Catalog API recently, but it was too difficult for my use case, while Infracost API was a pleasure to use, so please check out #36.
